ansible/roles/operator-pipeline/templates/openshift/tasks/merge-registry-credentials.yml

```diff
@@ -0,0 +1,56 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: merge-registry-credentials
+spec:
+  params:
+    - name: cert_project_docker_config_json
+      description: Encrypted project's docker_config_json that contains private
+        registry tokens
+  workspaces:
+    - name: registry-credentials
+      description: Registry tokens provided as an input of a pipeline
+    - name: registry-credentials-all
+      description: Global workspace for storing all registry access tokens
+    - name: gpg-key
+      description: GPG private key that is used for decryption of user's tokens
+  steps:
+    - name: get-and-decrypt-project-tokens
+      image: quay.io/redhat-isv/operator-pipelines-images:latest
+      workingDir: $(workspaces.registry-credentials-all.path)
+      script: |
+        #! /usr/bin/env bash
+        set -e
+
+        export GPG_TTY=$(tty)
+        export GNUPGHOME=/tmp
+
+        PROJECT_CONFIG_JSON="$(params.cert_project_docker_config_json)"
+        echo $PROJECT_CONFIG_JSON
+
+        if [[ $PROJECT_CONFIG_JSON != "null" ]]; then
+
+          echo "Project contains secret registry tokens. Merging with internal tokens..."
+          echo $PROJECT_CONFIG_JSON | sed 's/\\n/\n/g' > project_dockerconfigjson.gpg
+
+          gpg -v --batch --import <(cat "$(workspaces.gpg-key.path)/exd-guild-isv.key")
+          GPG_PASSPHRASE=`(cat "$(workspaces.gpg-key.path)/exd-guild-isv.passphrase")`
+
+          gpg \
+            --output project_dockerconfigjson \
+            --batch \
+            --yes \
+            --passphrase $GPG_PASSPHRASE \
+            --pinentry-mode=loopback \
+            --decrypt project_dockerconfigjson.gpg
+
+          jq -s '.[0] * .[1]' \
+            $(workspaces.registry-credentials.path)/.dockerconfigjson \
+            project_dockerconfigjson \
+            > "$(workspaces.registry-credentials-all.path)/.dockerconfigjson"
+          echo "Tokens successfully merged"
```

Comment:
Should there be some sort of check before assuming that the merge was successful? Maybe verifying that the result is a valid json, or at least checking to make sure the last command finished with code 0.